File: metodos/Newton.py
from math import*
import numpy as np

def Newton(f, derf, x0, tol, Nmax):
    xant = x0
    flambda = lambda x: eval(f)
    fant = flambda(xant)
    E = 1000
    cont = 0
    derflambda = lambda x: eval(derf)
    while E > tol and cont < Nmax:
      xact = xant-fant/(derflambda(xant))
      fact = flambda(xact)
      E = abs(xact-xant)
      cont = cont+1
      xant = xact
      fant = fact
      if E < tol
        x = xact
        ite = cont
        err = E
        print("La función tiene una raíz en, " x, "con una tolerancia de, " E, "y un número de iteraciones de " ite)
      else
        print("No tiene solucion")

File: metodos/Newton.java
import java.util.function.DoubleUnaryOperator;

public class Newton {
    public static double newton(DoubleUnaryOperator f, DoubleUnaryOperator derf, double x0, double tol, int Nmax) {
        double xant = x0;
        double fant = f.applyAsDouble(xant);
        double E = 1000;
        int cont = 0;

        while (E > tol && cont < Nmax) {
            double xact = xant - fant / derf.applyAsDouble(xant);
            double fact = f.applyAsDouble(xact);
            E = Math.abs(xact - xant);
            cont++;
            xant = xact;
            fant = fact;
            if (E < tol) {
                return xact;
            }
        }

        throw new RuntimeException("No se encontró una raíz en el rango especificado.");
    }
}
